OrbitCore/Callstack.h

```diff
@@ -7,20 +7,24 @@
 
 #include <xxhash.h>
 
-#include <string>
 #include <vector>
 
 #include "CallstackTypes.h"
 
 struct CallStack {
   CallStack() = default;
-  inline CallstackID Hash() {
-    if (m_Hash != 0) return m_Hash;
-    m_Hash = XXH64(m_Data.data(), m_Data.size() * sizeof(uint64_t), 0xca1157ac);
-    return m_Hash;
-  }
+  explicit CallStack(std::vector<uint64_t>&& addresses) {
+    data = std::move(addresses);
+    hash = XXH64(data.data(), data.size() * sizeof(uint64_t), 0xca1157ac);
+  };
 
-  CallstackID m_Hash = 0;
-  std::vector<uint64_t> m_Data;
+  CallstackID GetHash() const { return hash; }
+  uint64_t GetData(size_t id) const { return data.at(id); }
```

Comment:
I would be more specific with naming here: `GetData` could be `GetFrame` or `GetIp` or `GetPc` and `id` could be `index`.